### PR TITLE
Add migration guide for platform 0.9

### PR DESCRIPTION
**Pull request by natalieparellano**

### CODE REVIEW

**Comment by natalieparellano** [comment]
@buildpacks/learning-maintainers is it possible to get a review on this?

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
Let's merge this once the new lifecycle release is out. Marking it as draft until then.

**Pull request merged**
